File: src/main/java/spark/Dataframe6.java
package spark;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

import java.util.Collections;

import static org.apache.spark.sql.functions.*;

public class Dataframe6 {
    public static void main(String[] args) {

        SparkSession spark = SparkSession
                .builder()
                .appName("Java Spark SQL basic example").master("local")
                .getOrCreate();

        Dataset<Row> ratings = spark.read().format("csv").option("header", "true")
                .load("/Users/navom_saxena/Downloads/ratings.csv");

        Dataset<Row> movies = spark.read().format("csv").option("header", "true")
                .load("/Users/navom_saxena/Downloads/movies.csv");

        Dataset<Row> joinedDf = ratings.join(movies, scala.collection.JavaConverters.asScalaIteratorConverter(
                Collections.singletonList("movieId").iterator()
        ).asScala().toSeq(), "inner");

        joinedDf.groupBy("title")
                .agg(max("rating"), min("rating"), avg("rating"))
                .show();

    }
}

File: src/main/python/pyspark/Dataframe6.py
from pyspark.sql import SparkSession
from pyspark.sql.functions import max, min


class MoviesJoin:
    if __name__ == '__main__':
        spark = SparkSession.builder.getOrCreate()
        df = spark.read.format('csv').option("header", "true").load("/Users/navom_saxena/Downloads/ratings1.csv");

        df2 = spark.read.format('csv').option("header", "true").load("/Users/navom_saxena/Downloads/movies.csv");

        inner_join = df1.join(df2, df.movieId == df2.movieId).groupBy("title").agg(max("rating"), min("rating"),
                                                                                   avg("rating")).show()
